cns/restserver/util.go

```diff
@@ -923,3 +923,60 @@ func (service *HTTPRestService) createNetworkContainers(createNetworkContainerRe
 		Message:    "",
 	}
 }
+
+// setResponse encodes the http response
+func (service *HTTPRestService) setResponse(w http.ResponseWriter, returnCode types.ResponseCode, response interface{}) {
+	serviceErr := service.Listener.Encode(w, &response)
+	logger.Response(service.Name, response, returnCode, serviceErr)
+}
+
+// generateAndCacheGetHomeAzResponse build the GetHomeAzResponse and cache it
+func (service *HTTPRestService) generateAndCacheGetHomeAzResponse(returnMessage string, returnCode types.ResponseCode, homeAzResponse nmagent.HomeAzResponse) cns.GetHomeAzResponse {
+	getHomeAzResponse := cns.GetHomeAzResponse{
+		HomeAzResponse: homeAzResponse,
+		Response:       cns.Response{ReturnCode: returnCode, Message: returnMessage},
+	}
+	service.updateGetHomeAzResponseCache(getHomeAzResponse)
+	return getHomeAzResponse
+}
+
+// readNMASupportedApisCache gets the nmsSupportedApisCache value
+func (service *HTTPRestService) readNMASupportedApisCache() []string {
+	service.RLock()
+	supportedApis := service.nmaSupportedApisCache
+	service.RUnlock()
```

Comment:
I don't think you need to lock the whole service for this, do you? We should have smaller scoped mutexes for things which can be safely accessed in parallel with each other.
/cc @timraymond again